integration/localnet/bootstrap.go

```diff
@@ -439,13 +460,14 @@ func prepareExecutionService(container testnet.ContainerConfig, i int) Service {
 	service.Ports = []string{
 		fmt.Sprintf("%d:%d", ExecutionAPIPort+2*i, RPCPort),
 		fmt.Sprintf("%d:%d", ExecutionAPIPort+(2*i+1), SecuredRPCPort),
+		fmt.Sprintf("%d:%d", AdminToolLocalPort+n, AdminToolPort),
```

Comment:
Instead of the `+n` mechanism, why not use the `container.ContainerName` mechanism like `--ingress-addr` above? This way there is no need for the `+n` and the name and port is more predictable regardless of localnet configuration? E.g. `fmt.Sprintf("--admin-addr=%s:%d", container.ContainerName, 9002),`? Pros? Cons?

Reply:
`fmt.Sprintf("%d:%d", AdminToolLocalPort+n, AdminToolPort)` will add a port mapping as `3702:9002`. 9002 is the internal port of that container, and `3702` will map that port to your local port, so that we can talk to the admin tool port `9002` via the local port `3702`. Since every node is using `9002` as admin tool port, and we only has one `9002` port, so we will map each node's 9002 to a different port without overlapping. 

If we use `container name`, how it works to access a port via container name. we can't send request to `curl http://127.0.0.1:localnet_consensus_1_1`, right?

Reply:
> If we use container name, how it works to access a port via container name. we can't send request to curl `http://127.0.0.1:localnet_consensus_1_1`, right?

I assumed we would send a curl request to `http://consensus_1:9000` and that docker would handle it. But now I'm assuming -- because of your answer below -- that this kind of domain only works inside the node group, thus making it not possible to send such a curl request outside the node group.

And this [1] from 2016 also confirms that docker hosts cannot see the internal docker network :-(

Sorry about misunderstanding and thinking that docker was more capable...

[1] https://stackoverflow.com/questions/36151981/local-hostnames-for-docker-containers